crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -625,10 +625,10 @@ where
         let mut prepared_stream = self.txs_stream.lock().await;
 
         while let Ok(tx) = prepared_stream.try_recv() {
-            pending_txs.push(from_recovered(
-                tx.transaction.to_recovered_transaction(),
-                &self.tx_resp_builder,
-            ))
+            pending_txs.push(
+                from_recovered(tx.transaction.to_recovered_transaction(), &self.tx_resp_builder)
+                    .expect("fill should be infallible"),
```

Comment:
this expect still needs to go

Reply:
I've removed the expect and tried logging the error so that it doesn't become nightmare to propagate the error.(I tried propagating it first but requires changing a lot of types).